language/move-prover/bytecode-to-boogie/test_mvir/verify-vector.mvir

```diff
@@ -294,4 +294,98 @@ module VerifyVector {
         y = Vector.borrow<u64>(&ev1, 0);
         return (0, *move(y));
     }
+
+    // succeeds.
+    public my_length1(v: Vector.T< u64>): u64
+    ensures vector_length(v) == RET(0)
+    // TODO: spec_translator cannot handle the following post-conditions
+    // ensures vector_length(v) + 1 == RET(0) // thread 'main' panicked at 'unexpected type', language/move-prover/bytecode-to-boogie/src/spec_translator.rs:343:18
```

Comment:
This crash is gone since a PR yesterday. 

But it was actually hiding an actual error message. There is a conceptual restriction with boogie helper functions. They can't be nested in expressions, because the spec translator does not know their result type. So you can't write `helper_fun() + 1`. You can only use them in relations.

This is something we may want to fix depending how long it takes until we have the real language extensions.